src/components/taskDetails/Details.tsx

```diff
@@ -1,20 +1,67 @@
 import React, { FC } from 'react';
+import moment from 'moment';
+import Link from 'next/link';
+import { FaReceipt } from 'react-icons/fa6';
+import Tooltip from '@/components/common/Tooltip/Tooltip';
+import { useRouter } from 'next/router';
 import setColor from './taskPriorityColors';
 import styles from './task-details.module.scss';
 import { TaskDetailsProps } from '@/interfaces/taskDetails.type';
 import extractRepoName from '@/utils/extractRepoName';
-import Link from 'next/link';
-import { FaReceipt } from 'react-icons/fa6';
-import DevFeature from '../DevFeature';
 
 const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
     const color = value ? setColor?.[value] : undefined;
     const isGitHubLink = detailType === 'Link';
     const gitHubIssueLink = isGitHubLink ? value : undefined;
+    const router = useRouter();
+    const isDevMode = router.query.dev === 'true' ? true : false;
+
+    const getRelativeTime = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).fromNow() : 'N/A';
+    };
+
+    const getTooltipText = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).format('LLLL') : 'N/A';
+    };
+
+    const isPastDate = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).isBefore(moment()) : false;
+    };
+
+    const getHumanReadableTime = (timestamp: string | number | undefined) => {
+        if (!timestamp) return 'N/A';
+
+        const now = moment();
+        const futureTimestamp = moment(timestamp);
+
+        if (futureTimestamp.isAfter(now)) {
+            return `in ${moment
+                .duration(futureTimestamp.diff(now))
+                .humanize()}`;
+        } else {
```

Comment:
- we do not need the else block here as we do an early return in the if clause.

Reply:
remove the `else` block from the code.
@yesyash please re-check